CHANGELOG.md

```diff
@@ -1,6 +1,20 @@
 # Change Log
 All notable changes to this project will be documented in this file.
 
+## \[UNRELEASED\]
+
+### Added
+
+### Fixed
+
+### Changed
+
+### Deprecated
+
+-   \[INTERNALS\] #1584 : Remove unused functions : `inline`, `subs`, `is_iterable_datatype`, `is_with_construct_datatype`, `is_pyccel_datatype`.
```

Comment:
I would specify the modules from which the functions have been removed. Further, I believe the function `get_iterable_ranges` is missing.